app/controllers/listings_controller.rb

```diff
@@ -0,0 +1,43 @@
+class ListingsController < ApplicationController
+  def index
+    @listing = Listing.all
+  end
+
+  def show
+    @song = Song.find(params[:id])
+  end
+
+  def new
+    @listing = Listing.new
+  end
+
+  def create
+    @listing = Listing.create(listing_params)
+    @listing.user = current_user
+    if @listing.save
+      redirect_to listing_path(@listing)
+    else
+      render :new
+    end
+  end
+
+  def update
+    @listing = Listing.new(listing_params)
+     if @listing.save!
+      redirect_to listing_path(@listing)
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def set_listing
+    @listing = Listing.find(params[:id])
+  end
+
+  def listing_params
+    params[:listings].permit(:name, :description, :status) #did not add :photo, :price: status. At least not yet.
```

Comment:
Should be `params.require(:listing).permit(:name, :description, :status)`

Reply:
Since our migration files have it as :listing, I think we should keep it that way.